newIDE/app/src/AssetStore/HexColorField.js

```diff
@@ -0,0 +1,115 @@
+// @flow
+import { t } from '@lingui/macro';
+import * as React from 'react';
+import TextField from '../UI/TextField';
+import ColorPicker, { type ColorResult } from '../UI/ColorField/ColorPicker';
+import {
+  type RGBColor,
+  hexToRGBColor,
+  rgbColorToHex,
+} from '../Utils/ColorTransformer';
+
+const styles = {
+  container: {
+    position: 'relative',
+    display: 'inline-block',
+  },
+  picker: {
+    position: 'absolute',
+    right: '8px',
+    top: '19px',
+  },
+  pickerNoLabel: {
+    position: 'absolute',
+    right: '8px',
+    top: '8px',
+  },
+};
+
+type Props = {|
+  fullWidth?: boolean,
+  disableAlpha?: boolean,
+  id?: string,
+  floatingLabelText?: string | React.Node,
+  helperMarkdownText?: ?string,
+  onChange: (RGBColor | null) => void,
+  color: RGBColor | null,
+|};
+
+type State = {|
+  color: string,
+|};
+
+const hexToNullableRGBColor = (color: string): RGBColor | null => {
+  return /^#{0,1}[0-9a-fA-F]{6}$/.test(color) ? hexToRGBColor(color) : null;
+};
+
+export class HexColorField extends React.Component<Props, State> {
```

Comment:
Why not use a functional component? (especially for something we just created)

Reply:
It's a copy of an existing component and I wasn't sure it would be kept.

Reply:
I feel like we should keep it, like you mentioned, the usage is different, so we may end up modifying it in the future without affecting the other (and vice versa)

Reply:
Do you think a prop like `preferHexFormat={true}` would work to reuse ColorField?

Reply:
There are already a lot of color conversions, it might make it hard to understand.